packages/react-devtools-shared/src/constants.js

```diff
@@ -10,6 +10,9 @@
 // Flip this flag to true to enable verbose console debug logging.
 export const __DEBUG__ = false;
 
+// Flip this flag to true to enable performance logging to console.
+export const __PROFILE__ = false;
```

Comment:
Don't have time to review this full PR now, but a quick thought: The `__PROFILE__` flag has a specific meaning for React builds already, and I think this functionality is a bit different. Maybe we should use a different name to avoid confusion?